lib/midcom/helper/imagepopup/handler/upload.php

```diff
@@ -1,161 +1,169 @@
 <?php 
+/**
+ * @author lukasz chalat
+ * @package midcom.helper.imagepopup
+ * @copyright The Midgard Project, http://www.midgard-project.org
+ * @license http://www.gnu.org/licenses/lgpl.html GNU Lesser General Public License
+ */
 
 /**
- * This handler uploads the attachments attached to object $object.
+ * This handler get uploaded file and save it in database and write to file.
  *
  * @package midcom.helper.imagepopup
  */
 
 class midcom_helper_imagepopup_handler_upload extends midcom_baseclasses_components_handler
-{	
-	public function _handler_upload($handler_id, array $args, array &$data)
-	{
-		// Ensure we get the correct styles
-		midcom::get()->style->prepend_component_styledir('midcom.helper.imagepopup');
-		midcom::get()->skip_page_style = true;
-		
-		
-		// Get the file
-		reset ($_FILES);
-		$temp = current($_FILES);
-		
-		$accepted_origins = array("http://localhost", "http://openpsa/");
-		
-		if(is_uploaded_file($temp['tmp_name']))
-		{
-			if (isset($_SERVER['HTTP_ORIGIN']))
-			{
-				// same-origin requests won't set an origin. If the origin is set, it must be valid.
-				if (in_array($_SERVER['HTTP_ORIGIN'], $accepted_origins))
-				{
-					header('Access-Control-Allow-Origin: ' . $_SERVER['HTTP_ORIGIN']);
-				}
-				else
-				{
-					header("HTTP/1.0 403 Origin Denied");
-					return;
-				}
-			}
-		
-			if (preg_match("/([^\w\s\d\-_~,;:\[\]\(\).])|([\.]{2,})/", $temp['name']))
-			{
-				header("HTTP/1.0 500 Invalid file name.");
-				return;
-			}
-		
-			if (!in_array(strtolower(pathinfo($temp['name'], PATHINFO_EXTENSION)), array("gif", "jpg", "png")))
-			{
-				header("HTTP/1.0 500 Invalid extension.");
-				return;
-			}
-		}
-		
-		// Get the data
-		$temp_name = $temp['tmp_name'];
-		$mimetype = $temp['type'];
-		
-		switch($handler_id)
-		{
-			case '____ais-imagepopup-upload_image':
-				$parentguid = $args[0];
-				break;
-		}
-		
-		// Set modified filename 
-		$filename = $this->getModifyFilename($temp['name']);
-		
-		// Set the part of the location
-		$core = "/midcom-serveattachmentguid-";
-		
-		// Insert the image into database
-		$this->insertDatabase($filename, $mimetype, $parentguid);
-            
-        // Get guid of the image
-        $guid = $this->getDataFromDatabase($filename, "guid");
+{
+    /**
+     * @param mixed $handler_id The ID of the handler.
+     * @param array $args The argument list.
+     * @param array &$data The local request data.
+     */
+    public function _handler_upload($handler_id, array $args, array &$data)
+    {
+        // Ensure we get the correct styles
+        midcom::get()->style->prepend_component_styledir('midcom.helper.imagepopup');
+        midcom::get()->skip_page_style = true;
```

Comment:
You don't need these two lines if you return a JSON response, this ges set automatically by the framework in this case